napari/_qt/dialogs/qt_package_installer.py

```diff
@@ -0,0 +1,419 @@
+
+    # -------------------------- Private methods ------------------------------
+    def _log(self, msg: str):
+        log.debug(msg)
+        if self._output_widget:
+            self._output_widget.append(msg)
+
+    def _build_queue_item(
+        self,
+        task_handler: TaskHandler,
+        action: ActionType,
+        pkgs: Sequence[str],
+        prefix: Optional[str] = None,
+        origins: Sequence[str] = (),
+        **kwargs,
+    ) -> AbstractInstallerTask:
+        if task_handler == "pip":
+            InstallerAction = PipInstallerTask
+        elif task_handler == "conda":
+            InstallerAction = CondaInstallerTask
+        else:
+            raise ValueError(f"Handler {task_handler} not recognized!")
+        return InstallerAction(
+            pkgs=pkgs, action=action, origins=origins, prefix=prefix, **kwargs
+        )
+
+    def _queue_item(self, item: AbstractInstallerTask) -> JobId:
+        self._queue.append(item)
+        self._process_queue()
+        return item.ident
+
+    def _process_queue(self):
+        if not self._queue:
+            self.allFinished.emit()
+            return
+        task = self._queue[0]
+        self.setProgram(str(task.executable()))
+        self.setProcessEnvironment(task.environment())
+        self.setArguments([str(arg) for arg in task.arguments()])
+        # this might throw a warning because the same process
+        # was already running but it's ok
+        self._log(f"Starting '{self.program()}' with args {self.arguments()}")
+        self.start()
+
+    def _on_process_finished(
+        self, exit_code: int, exit_status: QProcess.ExitStatus
+    ):
+        try:
+            current = self._queue[0]
+        except IndexError:
+            current = None
+        if (
+            current
+            and current.action == "uninstall"
+            and exit_status == QProcess.ExitStatus.NormalExit
+            and exit_code == 0
+        ):
+            pm2 = PluginManager.instance()
+            npe1_plugins = set(plugin_manager.iter_available())
+            for pkg in current.pkgs:
+                if pkg in pm2:
+                    pm2.unregister(pkg)
+                elif pkg in npe1_plugins:
+                    plugin_manager.unregister(pkg)
+                else:
+                    log.warning(
+                        'Cannot unregister %s, not a known napari plugin.', pkg
+                    )
+        self._on_process_done(exit_code=exit_code, exit_status=exit_status)
+
+    def _on_error_occurred(self, error: QProcess.ProcessError):
+        self._on_process_done(error=error)
+
+    def _on_process_done(
+        self,
+        exit_code: Optional[int] = None,
+        exit_status: Optional[QProcess.ExitStatus] = None,
+        error: Optional[QProcess.ProcessError] = None,
+    ):
+        with contextlib.suppress(IndexError):
+            self._queue.popleft()
+        msg = "Task finished with "
```

Comment:
We need to add translation for strings 

See https://napari.org/stable/developers/translations.html

Reply:
@goanpeca - does this apply to the messages in raised `Exception`s too?